src/test/kotlin/nl/info/zac/app/zaak/ZaakKoppelenRestServiceTest.kt

```diff
@@ -160,6 +160,200 @@ class ZaakKoppelenRestServiceTest : BehaviorSpec({
+        }
+
+        val zoekResultaat = ZoekResultaat(listOf(zaakZoekObject), 1)
+
+        every { zrcClientService.readZaak(sourceZaak.uuid) } returns sourceZaak
+        every { searchService.zoek(any()) } returns zoekResultaat
+        every { policyService.readZaakRechten(sourceZaak).koppelen } returns true
+        every { policyService.readZaakRechten(zaakZoekObject).koppelen } returns true
+
+        When("findLinkableZaken with HOOFDZAAK link is called") {
+            val result = zaakKoppelenRestService.findLinkableZaken(
+                zaakUuid = sourceZaak.uuid,
+                zoekZaakIdentifier = zoekZaakIdentifier,
+                relationType = RelatieType.HOOFDZAAK,
+                page = page,
+                rows = rows
+            )
+
+            Then("a single linkable zaak should be returned") {
+                result.resultCount shouldBe 1
+            }
+
+            And("link is not possible") {
+                with(result.results.first()) {
+                    id shouldBe zaakZoekObject.getObjectId()
+                    type shouldBe zaakZoekObject.getType()
+                    identificatie shouldBe zoekZaakIdentifier
+                    omschrijving shouldBe OMSCHRIJVING
+                    zaaktypeOmschrijving shouldBe ZAAK_TYPE_OMSCHRIJVING
+                    statustypeOmschrijving shouldBe STATUS_TYPE_OMSCHRIJVING
+                    isKoppelbaar shouldBe false
+                }
+            }
+
+            And("required services should be invoked") {
+                verify(exactly = 1) {
+                    zrcClientService.readZaak(sourceZaak.uuid)
+                    searchService.zoek(any())
+                    policyService.readZaakRechten(sourceZaak)
+                    policyService.readZaakRechten(zaakZoekObject)
+                }
+            }
+        }
+
+        When("findLinkableZaken with DEELZAAK link is called") {
+            val result = zaakKoppelenRestService.findLinkableZaken(
+                zaakUuid = sourceZaak.uuid,
+                zoekZaakIdentifier = zoekZaakIdentifier,
+                relationType = RelatieType.DEELZAAK,
+                page = page,
+                rows = rows
+            )
+
+            Then("a single linkable zaak should be returned") {
+                result.resultCount shouldBe 1
+            }
+
+            And("link is not possible") {
+                with(result.results.first()) {
+                    id shouldBe zaakZoekObject.getObjectId()
+                    type shouldBe zaakZoekObject.getType()
+                    identificatie shouldBe zoekZaakIdentifier
+                    omschrijving shouldBe OMSCHRIJVING
+                    zaaktypeOmschrijving shouldBe ZAAK_TYPE_OMSCHRIJVING
+                    statustypeOmschrijving shouldBe STATUS_TYPE_OMSCHRIJVING
+                    isKoppelbaar shouldBe false
+                }
+            }
+
+            And("required services should be invoked") {
```

Comment:
```suggestion
            And("required services should've been invoked") {
```
and why 2 invocations on these services? maybe that could be included in the `And`

Reply:
That's a matter of IsolationMode for kotest. Will change this.